wallet/src/main/scala/org/bitcoins/wallet/internal/AddressHandling.scala

```diff
@@ -184,6 +184,63 @@ private[wallet] trait AddressHandling extends WalletLogger {
     addrF
   }
 
+  def getAddress(
+      account: AccountDb,
+      chainType: HDChainType,
+      addressIndex: Int): Future[AddressDb] = {
+
+    val coinType = account.hdAccount.coin.coinType
+    val accountIndex = account.hdAccount.index
+
+    val path = account.hdAccount.purpose match {
+      case HDPurposes.Legacy =>
+        LegacyHDPath(coinType, accountIndex, chainType, addressIndex)
+      case HDPurposes.NestedSegWit =>
+        NestedSegWitHDPath(coinType, accountIndex, chainType, addressIndex)
+      case HDPurposes.SegWit =>
+        SegWitHDPath(coinType, accountIndex, chainType, addressIndex)
+    }
+
+    val pathDiff =
+      account.hdAccount.diff(path) match {
+        case Some(value) => value
+        case None =>
+          throw new RuntimeException(
```

Comment:
nit: Should probably be an `IllegalArgumentException` since this depends only on inputs (I think)